src/database/queries.ts

```diff
@@ -105,11 +150,50 @@ export const getVoucher = async (uuid: Uuid): Promise<Voucher> => {
   }
 };
 
-export const createVoucher = async (voucher: VoucherCreate): Promise<Uuid> => {
+/**
+ * Query to create a new voucher in Firebase.
+ *
+ * Parameters: a json with fields
+ *
+ *    `serialNumber`: voucher serial number
+ *
+ *    `value`: monetary value of voucher in cents
+ *
+ *    `vendorUuid`: current user's Uuid
+ *
+ * Returns the doc id if the query executes successfully.
+ *
+ * Returns `'-1'` if the serial number is invalid,
+ * `'-2'` if the serial number has already been used,
+ * or `'-3'` if the voucher's value exceeds the maximum possible value.
+ */
+export const createVoucher = async (
+  voucher: VoucherCreate,
+): Promise<string> => {
   try {
-    const docRef = await addDoc(voucherCollection, voucher);
-    await updateDoc(docRef, { uuid: docRef.id });
-    return docRef.id;
+    const docId = voucher.serialNumber.toString();
+    const docRef = doc(db, 'vouchers', docId);
+
+    // check that serialNumber exists
+    const voucherRange = await getVoucherRange(voucher.serialNumber);
+    if (voucherRange === -1) {
+      return '-1';
+    }
+
+    // check that serialNumber has not already been used
+    const voucherDoc = await getDoc(docRef);
+    if (voucherDoc.exists()) {
+      return '-2';
+    }
+
+    // check that value does not exceed the maximum
+    if (voucherRange.maxValue < voucher.value) {
+      return '-3';
+    }
+
+    await setDoc(docRef, voucher);
+    await updateDoc(docRef, { type: voucherRange.type });
```

Comment:
Looks like with this scheme, we're no longer using `uuid`s for each record, and instead we're using the serial number as a uuid. It seems to make sense given the assumption that serial numbers will always be unique, but I just wanted to flag it to make sure, since it strays from the existing convention. cc @sauhardjain

Reply:
Yeah, the purpose of doing this was to make sure look-ups for specific vouchers would be as quick as possible. 

@sauhardjain tagging myself to make sure I take care of this through Retool.

Reply:
we decided to make the change because the voucher database is expected to get very large, so using the serial number as the uuid should make lookups faster.